ledger/acctupdates.go

```diff
@@ -776,10 +777,42 @@ func accountHashBuilder(addr basics.Address, accountData basics.AccountData, enc
 }
 
 // Initialize accounts DB if needed and return account round
-func (au *accountUpdates) accountsInitialize(tx *sql.Tx) (basics.Round, error) {
-	err := accountsInit(tx, au.initAccounts, au.initProto)
+func (au *accountUpdates) accountsInitialize(ctx context.Context, tx *sql.Tx) (basics.Round, error) {
+	// check current database version.
+	dbVersion, err := db.GetUserVersion(ctx, tx)
 	if err != nil {
-		return 0, err
+		return 0, fmt.Errorf("accountsInitialize unable to read database schema version : %v", err)
+	}
+
+	// if database version is greater than supported by current binary, write a warning. This would keep the existing
+	// fallback behaviour where we could use an older binary iff the schema happen to be backward compatible.
+	if dbVersion > accountDBVersion {
+		au.log.Warnf("accountsInitialize database schema version is %d, but algod supports only %d", dbVersion, accountDBVersion)
+	}
+
+	if dbVersion < accountDBVersion {
+		au.log.Infof("accountsInitialize upgrading database schema from version %d to version %d", dbVersion, accountDBVersion)
+
+		for dbVersion < accountDBVersion {
+			au.log.Infof("accountsInitialize performing upgrade from version %d", dbVersion)
+			// perform the initialization/upgrade
+			switch dbVersion {
+			case 0:
+				dbVersion, err = au.upgradeDatabaseSchema0(ctx, tx)
+				if err != nil {
+					return 0, err
+				}
+			case 1:
+				dbVersion, err = au.upgradeDatabaseSchema1(ctx, tx)
+				if err != nil {
+					return 0, err
```

Comment:
how about logging more details here for the user?
Like DB upgrade failed. Possible next steps:
1. Remove data dir and catchup from scratch.
2. If you are relay runner, please contact Algorand and provide tracker.db for analysis.

Reply:
I will add some more comments and logging in a subsequent PR. I'd like to be cautious about instructing people to send us files. Instead, I'd rather that this would come from an Algorand representative directly.
( we don't want third party attempting to contact algorand and sending us files.. )